staging/src/k8s.io/code-generator/cmd/validation-gen/output_tests/unions/zz_generated.validations.go

```diff
@@ -0,0 +1,240 @@
+	errs = append(errs, validate.Enum(fldPath, in, d3Symbols)...)
+	return errs
+}
+
+var dU1UnionMembership = validate.NewDiscriminatedUnionMembership("d1")
+
+func Validate_DU1(in *DU1, fldPath *field.Path) (errs field.ErrorList) {
+	errs = append(errs, validate.DiscriminatedUnion(fldPath, in, dU1UnionMembership, in.D1)...)
+	// TypeMeta
+
+	// D1
+	errs = append(errs, Validate_D1(&in.D1, fldPath.Child("d1"))...)
+
+	return errs
+}
+
+var dU2UnionMembership = validate.NewDiscriminatedUnionMembership("d2", "m1", "m2")
+
+func Validate_DU2(in *DU2, fldPath *field.Path) (errs field.ErrorList) {
+	errs = append(errs, validate.DiscriminatedUnion(fldPath, in, dU2UnionMembership, in.D2, in.M1, in.M2)...)
+	// TypeMeta
+
+	// D2
+	errs = append(errs, Validate_D2(&in.D2, fldPath.Child("d2"))...)
+
+	// M1
+	if in.M1 != nil {
+		errs = append(errs, Validate_M1(in.M1, fldPath.Child("m1"))...)
+	}
+
+	// M2
+	if in.M2 != nil {
+		errs = append(errs, Validate_M2(in.M2, fldPath.Child("m2"))...)
+	}
+
+	// T1
+	if in.T1 != nil {
+		errs = append(errs, Validate_T1(in.T1, fldPath.Child("t1"))...)
+	}
+
+	return errs
+}
+
+var dU3UnionMembership = validate.NewDiscriminatedUnionMembership("d2", [2]string{"m1", "CustomM1"}, [2]string{"m2", "CustomM2"})
+
+func Validate_DU3(in *DU3, fldPath *field.Path) (errs field.ErrorList) {
+	errs = append(errs, validate.DiscriminatedUnion(fldPath, in, dU3UnionMembership, in.D3, in.M1, in.M2)...)
+	// TypeMeta
+
+	// D3
+	errs = append(errs, Validate_D3(&in.D3, fldPath.Child("d2"))...)
+
+	// M1
+	if in.M1 != nil {
+		errs = append(errs, Validate_M1(in.M1, fldPath.Child("m1"))...)
+	}
+
+	// M2
+	if in.M2 != nil {
+		errs = append(errs, Validate_M2(in.M2, fldPath.Child("m2"))...)
+	}
+
+	// T1
+	if in.T1 != nil {
+		errs = append(errs, Validate_T1(in.T1, fldPath.Child("t1"))...)
+	}
+
+	return errs
+}
+
+func Validate_M1(in *M1, fldPath *field.Path) (errs field.ErrorList) {
+	errs = append(errs, validate.FixedResult(fldPath, in, true, "type M1")...)
+	// S
+	errs = append(errs, validate.FixedResult(fldPath.Child("s"), in.S, true, "field M1.S")...)
+
+	return errs
+}
+
+func Validate_M2(in *M2, fldPath *field.Path) (errs field.ErrorList) {
+	errs = append(errs, validate.FixedResult(fldPath, in, true, "type M2")...)
+	// S
+	errs = append(errs, validate.FixedResult(fldPath.Child("s"), in.S, true, "field M2.S")...)
+
+	return errs
+}
+
+func Validate_T1(in *T1, fldPath *field.Path) (errs field.ErrorList) {
+	errs = append(errs, validate.FixedResult(fldPath, in, true, "type T1")...)
+	// TypeMeta
+
+	// LS
+	errs = append(errs, validate.FixedResult(fldPath.Child("ls"), in.LS, true, "field T1.LS")...)
+	for i, val := range in.LS {
+		errs = append(errs, validate.FixedResult(fldPath.Index(i), val, true, "field T1.LS[*]")...)
+		errs = append(errs, validate.Required(fldPath.Index(i), val)...)
+	}
+
+	return errs
+}
+
+var u1UnionMembership = validate.NewUnionMembership("m1", "m2")
```

Comment:
As with enums, I wonder if `sync.Once()` would let us move this inside the function

Reply:
This was a stupid suggestion